### PR TITLE
Shepherd tour

### PR DESCRIPTION
**Pull request by seemantk**
@curran, this is not yet ready for merge -- it's more of a beach-head tour with one single stop.
Would love your input on where to put the stops.

### CODE REVIEW

**Comment by curran** [comment]
As for the stops, I think the following would work well:

 * Map
 * Year menu
 * Grid
 * Menus (Tabulus)
 * Tabs

File: css/shepherd/shepherd-theme-square.css [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  border: 0;
+  margin: 0 .5em 0 0;
+  font-family: inherit;
+  text-transform: uppercase;
+  letter-spacing: .1em;
+  font-size: .8em;
+  line-height: 1em;
+  padding: .75em 2em;
+  background: #3288e6;
+  color: #fff; }
```
Comment (curran) [inline css/shepherd/shepherd-theme-square.css]: Somehow it doesn't feel quite right to include these vendor libs in the repo. Maybe we could point to unpkg CDN instead?

File: js/introModal.js [hunk 2]
```diff
@@ -12,6 +12,25 @@ function triggerIntroModal(){
     // is visiting the page for the first time.
     if(!pageSeen){
 
+
+        tour = new Shepherd.Tour({
```
Comment (curran) [inline js/introModal.js]: I thought the idea was that we keep the intro modal, and add a button there to start the tour?

**Comment by seemantk** [comment]
cheers -- I've disabled the popup here for local development.  I'll re-enable before removing the WIP label from this PR...

File: css/shepherd/shepherd-theme-square.css (hunk 1 above)
Comment (seemantk) [inline css/shepherd/shepherd-theme-square.css]: I'm all for it.  I had those in there for offline developing.

**Comment by seemantk** [comment]
@curran this is ready for your review.  I'm not currently sure how to add the button to the intro modal, but I'll have a look in the next day or so.

File: js/introModal.js [hunk 3]
```diff
@@ -21,11 +22,32 @@ function triggerIntroModal(){
...
+    // Run the tour always, until a "Start Tour" button is added to intro-popup.
+    tour = new Shepherd.Tour({
+        defaults: {
+              classes: 'shepherd-theme-arrows'
+            , scrollTo: true
+          }
+      })
+    ;
+  {% comment %}Create the tour using Jekyll{% endcomment %}
+  {% for stop in site.data.tour %}
```
Comment (curran) [inline js/introModal.js]: Very cool Jekyllization!

**Comment by curran** [comment]
@seemantk I've just added a new commit to this branch, which adds the "Take the tour" button to this intro modal and wires it up.

I think this main thing blocking mergeability here is that there's no way to exit the tour at the end. Besides that, I think this is ready to present as a first pass at the Shepherd tour.

There are also a few minor issues I'd like to address later in a second pass:

 * The first stop I think should introduce the whole concept of the map, its colors, and the tooltips.
 * Font size is quite small, smaller than the other text on the page. It would be nice to increase the font size in the tour boxes.
 * There's no wrapping or line breaks, so sometimes we end up with awkwardly wide tour stops. For example, this one could be broken into two lines:
image

File: js/introModal.js [hunk 4]
```diff
@@ -13,41 +39,32 @@ function triggerIntroModal(){
...
         // Track that the current user has visited this page using cookies.
         document.cookie = "pageSeen=true;expires=" + expiryDate.toUTCString();
+
+        // Add an event listener to the "Take tour" button after the modal loads.
+        // `loaded` event documented at http://getbootstrap.com/javascript/#modals-events
+        introModal.on("loaded.bs.modal", function (e) {
+            d3.select("#take-tour-button")
+                .on("click", function (event) {
+                    introModal.modal("hide");
+                    takeTour();
```
Comment (seemantk) [inline js/introModal.js]: Fantastic cleanup here @curran. Thanks for that!

File: js/introModal.js [hunk 5]
```diff
@@ -43,8 +43,8 @@ function triggerIntroModal(){
         , attachTo: '{{ stop.node }} {% if stop.orientation %}{{ stop.orientation }}{% else %}bottom{% endif %}'
         , buttons: [
               {
-                  text: {% unless forloop.last %}'Next'{% else %}'Start Over'{% endunless %}
-                , action: {% unless forloop.last %}tour.next{% else %}tour.show('step1'){% endunless %}
+                  text: {% unless forloop.last %}'Next'{% else %}'End Tour'{% endunless %}
+                , action: {% unless forloop.last %}tour.next{% else %}tour.complete{% endunless %}
```
Comment (curran) [inline js/introModal.js]: Nice!

**Pull request merged**